### PR TITLE
Add dhcp router handling for first/last/user defined IP

### PR DESCRIPTION
**Pull request by ccmarris**
This includes the tfilter updates, fixes for handling ansible variables in json parameters (e.g. address), clean-up in handling various address handling including special handling of routers where you can define the use of the 'first' IP in a subnet, the 'last' IP in the subnet or a user defined IP.

### CODE REVIEW

**Comment by ccmarris** [comment]
I've sorted the conflicts, can you please review?

File: ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py [hunk 1]
```diff
@@ -285,8 +285,6 @@ def main():
         host=dict(required=True, type='str'),
         address=dict(type='str'),
         space=dict(type='str'),
-        comment=dict(type='str'),
-        tags=dict(type='list', elements='dict', default=[{}]),
```
Comment (sriram-kannan-infoblox) [inline ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py:289]: May I know why we are removing the two field.

File: ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_subnet.py [hunk 2]
```diff
@@ -86,6 +88,7 @@
     tags:
       - {{ key }}: "{{ value }}"
     dhcp_options:
+      - routers: '<first|last|<IP Address>>'
```
Comment (sriram-kannan-infoblox) [inline ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_subnet.py:91]: We should create a sample playbook for the same.

Comment (ccmarris) [reply ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_subnet.py:91]: Certainly - I have a complex set of playbooks that use all of these features.

File: ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py (hunk 1 above)
Comment (ccmarris) [reply ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py:289]: Let me check - this may be in error/accidental

**Comment by ccmarris** [comment]
If the request came from Glenn, then that was me ;)

File: ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py (hunk 1 above)
Comment (ccmarris) [reply ansible_collections/infoblox/b1ddi_modules/plugins/modules/b1_ipam_address_block.py:289]: I think these were removed by accident, these lines should remain.

**Comment by ccmarris** [comment]
Can this be reviewed again, is there anything else I need to modify?

**Review by sriram-kannan-infoblox — APPROVED** [review]

**Pull request merged**